### PR TITLE
don't use IPv6 in tests when it's not available

### LINKED ISSUE #568: tests are stuck when running `make test`
I tried to use google's public dns as suggested by @sigmavirus24, but it didn't help.
```shell
$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 brd 127.255.255.255 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp5s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    inet 192.168.0.17/24 brd 192.168.0.255 scope global enp5s0
       valid_lft forever preferred_lft forever
    inet6 fe80::beae:c5ff:fe65:dd71/64 scope link 
       valid_lft forever preferred_lft forever
$ cat /etc/resolv.conf
nameserver 8.8.8.8
nameserver 8.8.4.4
```
strace:
```shell
$ strace -p 12676
Process 12676 attached
futex(0xbf8dd0, FUTEX_WAIT_PRIVATE, 0, NULL
```
This is gentoo, wicd and python 2.7. I think I had the same problem with Fedora and NetworkManager.

Also, last couple of lines from output:
```
test_oldapi (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_proxy_conn_fail (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_proxy_pooling (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_proxy_pooling_ext (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_proxy_verified (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_redirect (test.with_dummyserver.test_proxy_poolmanager.TestHTTPProxyManager) ... ok
test_multi_setcookie (test.with_dummyserver.test_socketlevel.TestCookies) ... Exception in thread Thread-8:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/home/tt/dev/urllib3/dummyserver/server.py", line 76, in run
    self.server = self._start_server()
  File "/home/tt/dev/urllib3/dummyserver/server.py", line 63, in _start_server
    sock.bind((self.host, 0))
  File "/usr/lib64/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
error: getsockaddrarg: bad family
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9549895-tests-are-stuck-when-running-make-test?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes #568

### CODE REVIEW

**Comment by Lukasa** [comment]
:+1: LGTM.

File: urllib3/exceptions.py [hunk 1]
```diff
@@ -144,6 +144,11 @@ class ResponseError(HTTPError):
     SPECIFIC_ERROR = 'too many {status_code} error responses'
 
 
+class NoIPv6Warning(HTTPWarning):
```
Comment (shazow) [inline urllib3/exceptions.py]: Can we move this somewhere under test/? I'd rather not pollute the core urllib3 code with test-related things. :)

Comment (TomasTomecek) [inline urllib3/exceptions.py]: sounds good; rebased and fixed

**Comment by TomasTomecek** [comment]
Fixed @shazow's suggestion (my comment is hidden in the *outdated diff* thing). Ready to get merged from my side.

**Comment by Lukasa** [comment]
Still LGTM. :+1:

**Comment by shazow** [comment]
Excellent, thank you!

**Pull request merged**